src/instamatic/camera/videostream.py

```diff
@@ -110,25 +124,81 @@ def __init__(self, cam: Union[CameraBase, str] = 'simulate'):
         self.name = self.cam.name
 
         self.frametime = self.default_exposure
-        self.frame = None
 
-        self.grabber = self.setup_grabber()
+        self.frame = NotImplemented
+        self.grabber = NotImplemented
 
-        self.streamable = self.cam.streamable
-
-        self.start()
-
-    def __getattr__(self, attrname):
+    def __getattr__(self, attr_name: str) -> Any:
         """Pass attribute lookups to self.cam to prevent AttributeError."""
         try:
-            return object.__getattribute__(self, attrname)
+            return object.__getattribute__(self, attr_name)
         except AttributeError as e:
             reraise_on_fail = e
+            if attr_name == 'cam':
+                return
             try:
-                return getattr(self.cam, attrname)
+                return getattr(self.cam, attr_name)
             except AttributeError:
                 raise reraise_on_fail
 
+    def close(self):
+        pass
+
+    def block(self):
+        pass
+
+    def unblock(self):
+        pass
+
+    @contextmanager
+    def blocked(self):
+        yield
+
+    def continuous_collection(self, exposure=0.1, n=100, callback=None):
+        """Function to continuously collect data Blocks the videostream while
+        collecting data, and only shows collected images.
+
+        exposure: float
+            exposure time
+        n: int
+            number of frames to collect
+            if defined, returns a list of collected frames
+        callback: function
+            This function is called on every iteration with the image as first argument
+            Should return True or False if data collection is to continue
+        """
+        buffer = []
+
+        go_on = True
+        i = 0
+
+        self.block()
+        while go_on:
+            i += 1
+
+            img = self.get_image(exposure=exposure)
+
+            if callback:
+                go_on = callback(img)
+            else:
+                buffer.append(img)
+                go_on = i < n
+
+        self.unblock()
+
+        if not callback:
+            return buffer
```

Comment:
+1 on deleting `continuous_collection()`, this was me playing around with some new ideas for the code. Probably not relevant anymore.

```suggestion
```